### PR TITLE
Performance improvement; convenience degradation.

### PR DESCRIPTION
**Pull request by shanecelis**
Hi Josh,

Nice work on your project. I have a rather large game project and when I first tried using UnityDomainReloadHelper the slow down was significant. It took 5 seconds according to the profiler.

I made some changes that greatly improved performance (down to 46 ms on my project), but it wasn't without costs. For instance, my fork no longer supports clearing properties or events. 

Because this is a breaking change, I'd totally understand not accepting this pull request. If you decide not to accept this pull request, perhaps I should pick a different name for this fork to make their differences clear.

Kind regards,

Shane

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,68 @@
...
+
+## License
+
+This project is released under the MIT license.
+
+## Acknowledgments
+
+The [original project](https://github.com/joshcamas/unity-domain-reload-helper) was written by Josh Steinhauer. Kudos on the design. I didn't know about TypeCache either, and I wrote similar assembly walking code that's still present in [Minibuffer Console](http://seawisphunter.com/products/minibuffer/).
+
+[^1]: This is not the original project. It was forked by me. See Changes section for more details.
```
Comment (joshcamas) [inline README.md:68]: Wanna remove this last line for the merge?

**Review by joshcamas — CHANGES_REQUESTED** [review]
Great work! I think just supporting fields is very much worth the gain! Could you remove the screenshots? After that it looks good to merge

**Comment by joshcamas** [comment]
Does typecache support static classes? I assume so?

File: README.md (hunk 1 above)
Comment (shanecelis) [reply README.md]: Happily. I adjusted the README to fit with the original.

**Comment by shanecelis** [comment]
I believe it does support static classes. 

I'm actually curious about whether TypeCache supports finding classes including in DLLs. My guess would be no, but I'd like to find out.

**Comment by Artein** [comment]
Is it possible to have a define to enable ability to "clear" properties/events after this PR (even with performance loss)?

**Comment by shanecelis** [comment]
I can see one way to do this without losing all the speed gains: Add an attribute for classes, say, `ScanForClearAttribute`. For only those particular classes, do the slow scan for attributes on properties and events. I'd profile it though and see how much time it takes.

**Comment by joshcamas** [comment]
Ah, I was under the impression that the slow part was the loading of assemblies or something along those lines. Requiring an additional attribute on the class itself seems like a great middleground. I'll merge this and then update it to support this attribute solution, with a define just to ensure as much speed by default. I'll profile it when I'm less busy and consider if there are better alternatives

**Comment by joshcamas** [comment]
On second thought, I think just clearing values via the method is fine. Trying to implement a whole new feature involving attributes on classes makes this tool a bit too inconsistent for my taste.

**Pull request merged**